x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java

```diff
@@ -53,6 +54,20 @@
 
     static Optional<DeprecationIssue> checkModelSnapshot(ModelSnapshot modelSnapshot) {
         if (modelSnapshot.getMinVersion().before(Version.V_7_0_0)) {
+            StringBuilder details = new StringBuilder(String.format(
+                Locale.ROOT,
+                "model snapshot [%s] for job [%s] supports minimum version [%s] and needs to be at least [%s].",
+                modelSnapshot.getSnapshotId(),
+                modelSnapshot.getJobId(),
+                modelSnapshot.getMinVersion(),
+                Version.V_7_0_0));
+            if (modelSnapshot.getLatestRecordTimeStamp() != null) {
+                details.append(String.format(
+                    Locale.ROOT,
+                    " The model snapshot's latest record timestamp is [%s]",
```

Comment:
```suggestion
                    ". The model snapshot's latest record timestamp is [%s]",
```

Reply:
It looks like there's already a full stop at the end of the message in line 59, so we shouldn't need another at the beginning of this extension.

I do think there's a slight formatting problem though, as the message on line 59 doesn't begin with a capital letter and this extension doesn't end with a full stop.  So we have proper sentence structure at the join but not at the very beginning or end.

We want these to look consistent with other warnings when they're displayed in the upgrade assistant in Kibana. @alisonelizabeth do you have any advice on whether these messages should have capital letters and full stops to fit in with the others that Kibana displays?

Reply:
> It looks like there's already a full stop at the end of the message in line 59, so we shouldn't need another at the beginning of this extension.

Ah, right. I was looking at the wrong place.